### ISSUE TITLE
"Don't pad `emphasis` with inner-spaces" warning on bullet-list

### ISSUE THREAD

**Issue by half-shell** [wooorm/remark-lint#119]
Hello!
First of all, thanks for remark-lint, it is something i've been looking for a while, and I'm glad I have found it.

The issue i'm having is in the following case:
```
### Header
    * [link](https://mylink.com) This is my link
    * [anotherlink](https://otherli.nk) This one is better
```
Remark sends me a ```67:5-68:6    warning    Don't pad `emphasis` with inner spaces    no-inline-padding    remark-lint```

I can avoid it by doing
```
### Header
    * [link](https://mylink.com) This is my link

    * [anotherlink](https://otherli.nk) This one is better
```
or
```
### Header

    * [link](https://mylink.com) This is my link
    * [anotherlink](https://otherli.nk) This one is better
```
but since I only have the issue in the specific case (link in bullet list), I thought i'd report the issue.

**Comment by wooorm** [comment]
Cheers, thanks @half-shell! Could you let me know how you use remark and remark-lint? Are there any other plugins, or options, you’re using?

**Comment by half-shell** [comment]
@wooorm Yes, I forgot to mention it. I'm using the [remark-preset-lint-recommended](https://github.com/wooorm/remark-lint/tree/master/packages/remark-preset-lint-recommended) preset.
I have it in my `.remarkrc`, and I am using remark as follow:
```bash
remark myFile.md -u remark-lint
```

**Comment by tkruse** [comment]
I see the same message for a different problem (code blocks non indented inside bullet list):

```
$ remark --config-path remarkrc hello.md 1> /dev/null 
hello.md
  5:15-5:58  warning  Don’t pad `emphasis` with inner spaces  no-inline-padding
  6:21-6:45  warning  Don’t pad `emphasis` with inner spaces  no-inline-padding
```

hello.md:
```
# Hello world

* provide a wrapper function, for example:

    void f(int* p);   // old code: f() does not modify `*p`
    void f(const int* p) { f(const_cast<int*>(p); } // wrapper
```



package.json:
```
    "dependencies": {
        "remark": "^4.2.2",
        "remark-lint": "^4.2.0",
        "remark-lint-sentence-newline": "^2.0.0",
        "remark-validate-links": "^4.1.0"
    }
```
remarkc:
```
{
    "plugins": {
        "remark-lint": {
            "list-item-indent": false
         }
    }
}
```

**Comment by wooorm** [comment]
Sorry for the late response!

Both your issues occur because that’s not a bulleted list, and that’s not a code-block.

@half-shell If you want a list, make sure to use one, two, or three spaces as indentation, not four:

Not:

```markdown
    * [link](https://mylink.com) This is my link
    * [anotherlink](https://otherli.nk) This one is better
```

But:

```markdown
* [link](https://mylink.com) This is my link
* [anotherlink](https://otherli.nk) This one is better
```

@tkruse If you want code in a list, you need to indent at least four spaces from the list-item indent.

Not:

```markdown
# Hello world

* provide a wrapper function, for example:

    void f(int* p);   // old code: f() does not modify `*p`
    void f(const int* p) { f(const_cast<int*>(p); } // wrapper
```

...but:

```markdown
# Hello world

* provide a wrapper function, for example:

      void f(int* p);   // old code: f() does not modify `*p`
      void f(const int* p) { f(const_cast<int*>(p); } // wrapper
```

**Comment by tkruse** [comment]
So just to confirm, the list-item-indent setting died not change the parser, it only suppresses one kind of warning?

**Comment by wooorm** [comment]
Correct, it only warns (or not).
